src/Type/IntersectionType.php

```diff
@@ -221,28 +221,38 @@ public function acceptsWithReason(Type $otherType, bool $strictTypes): AcceptsRe
 	}
 
 	public function isSuperTypeOf(Type $otherType): TrinaryLogic
+	{
+		return $this->isSuperTypeOfWithReason($otherType)->result;
+	}
+
+	public function isSuperTypeOfWithReason(Type $otherType): AcceptsResult
 	{
 		if ($otherType instanceof IntersectionType && $this->equals($otherType)) {
-			return TrinaryLogic::createYes();
+			return AcceptsResult::createYes();
 		}
 
 		if ($otherType instanceof NeverType) {
-			return TrinaryLogic::createYes();
+			return AcceptsResult::createYes();
 		}
 
-		return TrinaryLogic::createYes()->lazyAnd($this->getTypes(), static fn (Type $innerType) => $innerType->isSuperTypeOf($otherType));
+		return AcceptsResult::createYes()->and(...array_map(static fn (Type $innerType) => $innerType->isSuperTypeOfWithReason($otherType), $this->types));
```

Comment:
Here we're loosing a `lazyAnd` into a `and`.

I dunno if the lazy could be kept on AcceptsResults but I saw similar lazy lost when `isAcceptedWithReasonBy` was introduced